solana/programs/axelar-solana-gateway/README.md

```diff
@@ -0,0 +1,76 @@
+| [Verifier Set Tracker](https://github.com/eigerco/solana-axelar/blob/bf3351013ccf5061aaa1195411e2430c67250ec8/solana/programs/axelar-solana-gateway/src/state/verifier_set_tracker.rs) | Tracks information about an individual verifier set | Relayer, when rotationg verifier sets; Relayer, when approving messages; | Solana does not have built-in infinitie size hash maps as storage variables, using PDA for each verifier set entry allows us to ensure that duplicate verifier sets never get created | Gateway |
+| [Signtautre Verification Session](https://github.com/eigerco/solana-axelar/blob/bf3351013ccf5061aaa1195411e2430c67250ec8/solana/programs/axelar-solana-gateway/src/state/signature_verification_pda.rs) | Tracks that all the signatures for a given payload batch get verified | Relayer uses this in the multi-tx message approval process, where each signature from a verifier is sent individually to the Gateway for verification | | Gateway |
+| [Incoming Message](https://github.com/eigerco/solana-axelar/blob/bf3351013ccf5061aaa1195411e2430c67250ec8/solana/programs/axelar-solana-gateway/src/state/incoming_message.rs) | Tracks the state of an individual GMP message (executed / approved + metadata). | Relayer - after all the signatures have been approved, each GMP message must be initialized individually as well, relayer takes care of that. Destination program will receive this PDA in its `execute` flow when receiving the payload | | Gateway |
+| [Message Payload](https://github.com/eigerco/solana-axelar/blob/bf3351013ccf5061aaa1195411e2430c67250ec8/solana/programs/axelar-solana-gateway/src/state/message_payload.rs) | Contains the raw payload of a message. Limited of up to 10kb. Directly linked to an `IncomingMessage` PDA. | Relayer will upload the raw payload to a PDA and after message execution (or failure of execution) will close the PDA regaining all the funds. Destination program will receive this PDA in its `execute` flow. | Solana tx size limitation prevents sending large payolads directly on chain, thus the payload is stored directly on-chain | Gateway; Relayer that created it can also close it |
+| [Validate Call](https://github.com/eigerco/solana-axelar/blob/bf3351013ccf5061aaa1195411e2430c67250ec8/solana/programs/axelar-solana-gateway/src/lib.rs#L286-L291) | This acts only as a signing PDA, never initialized; Gives permission to the destination program to set `IncomingMessage` status to `executed`; | Destination program will craft this when making the CPI call to the Gateway | Emulates `msg.sender` from Solidity | Destination program |
+
+
+### Expectations from the destination contract
+
+For a third party developer to build an integrartion with the Axelar Solana Gateway, the only expectation is for the contract to implement [`axelar-executable`](../../crates/axelar-executable/README.md) interface. This allows the Relayer to have a known interfacte that it can send messages to, after they've been approved on the Gateway

```

Comment:
```suggestion
For a third party developer to build an integration with the Axelar Solana Gateway, the only expectation is for the contract to implement [`axelar-executable`](../../crates/axelar-executable/README.md) interface. This allows the Relayer to have a known interfacte that it can send messages to, after they've been approved on the Gateway

```